components-api/src/main/scala/pl/touk/nussknacker/engine/api/Service.scala

```diff
@@ -33,13 +33,52 @@ abstract class Service extends Lifecycle with Component
 // TODO: EagerService shouldn't extend Lifecycle, instead ServiceInvoker should extend it - see notice in ProcessCompilerData.lifecycle
 abstract class EagerService extends Service
 
-trait ServiceInvoker {
+trait ServiceLogic {
```

Comment:
I like this name, I see some :heart: to tapir, but this also introduces some diversity in naming.  We already have an ugly name for this concept. We call it "componentExecutor" and factory methods we call "implementation". See:
- `GenericNodeTransformation.implementation` - which something like `createLogic` factory method
- `ContextTransformation.implementedBy`
- our main domain object - `ComponentDefinitionWithImplementation` and `ComponentImplementationInvoker` referees to this `createLogic`
- `ComponentExecutorFactory` - in this case `ComponentExecutor` is just a logic.

If we introduce such a change IMO we should replace it everywhere.